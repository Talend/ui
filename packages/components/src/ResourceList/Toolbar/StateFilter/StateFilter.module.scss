@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;
@use '~@talend/design-tokens/lib/tokens';

$tc-resource-picker-toolbar-favorite-color: tokens.$coral-color-warning-text !default;
$tc-resource-picker-toolbar-certified-color: tokens.$coral-color-accent-text !default;

.tc-resource-picker-state-filters {
	margin-right: $padding-normal;
	display: flex;
	align-items: center;

	.option-label {
		margin-right: $padding-small;
	}

	// ActionIconToggle style reset :
	:global(.tc-icon-toggle) {
		border: none;
		box-shadow: none;

		> :global(.tc-icon) {
			margin: 0;
		}

		&:hover:not([disabled]),
		&:focus,
		&:active {
			border: none;
		}
	}

	:global(.tc-icon-toggle) {
		margin-left: $padding-small;
		display: inline-flex;

		&:global(.active),
		&:hover {
			&.tc-resource-picker-selection-filter > svg {
				color: tokens.$coral-color-accent-text;
			}

			&.tc-resource-picker-favorite-filter > svg {
				color: $tc-resource-picker-toolbar-favorite-color;
			}

			&.tc-resource-picker-certified-filter > svg {
				color: $tc-resource-picker-toolbar-certified-color;
			}
		}
	}
}
