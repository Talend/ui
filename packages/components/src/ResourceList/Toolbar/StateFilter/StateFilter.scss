$tc-resource-picker-toolbar-favorite-color: $lightning-yellow !default;
$tc-resource-picker-toolbar-certified-color: $brand-primary !default;

.tc-resource-picker-state-filters {
	margin-right: $padding-normal;
	display: flex;
	align-items: center;

	.option-label {
		margin-right: $padding-small;
	}

	// ActionIconToggle style reset :
	:global(.tc-icon-toggle) {
		background: none;
		border: none;
		box-shadow: none;
		width: $svg-md-size;

		> svg {
			margin: 0;
			width: $svg-md-size;
			height: $svg-md-size;
			color: $dove-gray;
		}

		&:hover:not([disabled]),
		&:focus,
		&:active {
			background: none;
			border: none;
		}
	}

	:global(.tc-icon-toggle) {
		margin-left: $padding-small;
		display: inline-flex;

		&:global(.active),
		&:hover {
			&.tc-resource-picker-selection-filter > svg {
				color: $scooter;
			}

			&.tc-resource-picker-favorite-filter > svg {
				color: $tc-resource-picker-toolbar-favorite-color;
			}

			&.tc-resource-picker-certified-filter > svg {
				color: $tc-resource-picker-toolbar-certified-color;
			}
		}
	}
}
