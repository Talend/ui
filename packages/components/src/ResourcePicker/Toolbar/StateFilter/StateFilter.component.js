import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { translate } from 'react-i18next';
import ActionIconToggle from '../../../Actions/ActionIconToggle';
import I18N_DOMAIN_COMPONENTS from '../../../constants';
import getDefaultT from '../../../translate';

import theme from './StateFilter.scss';

export const TYPES = {
	SELECTION: 'selection',
	FAVORITES: 'favorites',
	CERTIFIED: 'certified',
};

function StateFilter({ t, types, onChange, selection, favorites, certified }) {
	return (
		!!types.length && (
			<div
				className={classNames(
					'tc-resource-picker-state-filters',
					theme['tc-resource-picker-state-filters'],
				)}
			>
				<span className={classNames(theme['option-label'])}>
					{t('FILTER', { defaultValue: 'Filter:' })}
				</span>
				{types.includes(TYPES.SELECTION) && (
					<ActionIconToggle
						icon={'talend-check-circle'}
						label={t('SELECTION', { defaultValue: 'Selected' })}
						active={selection}
						onClick={() => onChange(TYPES.SELECTION, !selection)}
						className={classNames(theme['tc-resource-picker-selection-filter'])}
					/>
				)}
				{types.includes(TYPES.CERTIFIED) && (
					<ActionIconToggle
						icon={'talend-badge'}
						label={t('CERTIFIED', { defaultValue: 'Certified' })}
						active={certified}
						onClick={() => onChange(TYPES.CERTIFIED, !certified)}
						className={classNames(theme['tc-resource-picker-certified-filter'])}
					/>
				)}
				{types.includes(TYPES.FAVORITES) && (
					<ActionIconToggle
						icon={'talend-star'}
						label={t('FAVORITES', { defaultValue: 'Favorites' })}
						active={favorites}
						onClick={() => onChange(TYPES.FAVORITES, !favorites)}
						className={classNames(theme['tc-resource-picker-favorite-filter'])}
					/>
				)}
			</div>
		)
	);
}

StateFilter.propTypes = {
	t: PropTypes.func,
	selection: PropTypes.bool,
	favorites: PropTypes.bool,
	certified: PropTypes.bool,
	onChange: PropTypes.func,
	types: PropTypes.array,
};

StateFilter.defaultProps = {
	t: getDefaultT(),
	types: [TYPES.SELECTION, TYPES.FAVORITES, TYPES.CERTIFIED],
};

export default translate(I18N_DOMAIN_COMPONENTS)(StateFilter);
