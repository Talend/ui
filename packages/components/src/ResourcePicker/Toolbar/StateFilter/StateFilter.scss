.state-filters {
	margin-right: $padding-normal;

	> * {
		display: inline-block;
		height: 100%;
	}

	.option-label {
		margin-right: $padding-small;
	}

	// ActionIconToggle style reset :
	:global(.tc-icon-toggle) {
		background: none;
		border: none;
		box-shadow: none;
		width: $svg-md-size;

		> svg {
			margin: 0;
			width: $svg-md-size;
			height: $svg-md-size;
			color: $dove-gray;
		}

		&:hover:not([disabled]),
		&:active {
			background: none;
			border: none;
		}
	}

	:global(.tc-icon-toggle) {
		margin-left: $padding-small;
		display: inline-flex;

		&:global(.active),
		&:hover {
			&.selected-filter > svg {
				color: $scooter;
			}

			&.favorite-filter > svg {
				color: $lightning-yellow;
			}

			&.certified-filter > svg {
				color: $brand-primary;
			}
		}
	}
}
