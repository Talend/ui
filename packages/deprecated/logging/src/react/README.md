# React error logger

This module register a React updates hook that sends a log to an endpoint.

You need to configure 2 info
* the log server url
* the app state to send

## Configuration

You need to configure the logger in order to fit your backend logging service and add your app state.

```javascript
import { store as cmfstore } from '@talend/react-cmf';
import initReactLogger from '@talend/log/lib/react';

const store = cmfstore.initialize(appReducer);
initReactLogger({
    serverUrl: 'http://localhost:8888/error',
    getState: () => store.getState(),
    processState: state => removeSensitiveData(state),
});

```

You need to remove the user data from app state.
