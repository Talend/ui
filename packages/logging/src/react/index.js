import initReactLogger from './react-logger';

export default initReactLogger;
