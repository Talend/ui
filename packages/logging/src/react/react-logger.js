import ReactUpdates from 'react-dom/lib/ReactUpdates';
import ReactDefaultBatchingStrategy from 'react-dom/lib/ReactDefaultBatchingStrategy';
import { initErrorTransformer, TraceKit } from '../api/errorTransformer';
import getStatePayloadMiddleware from '../api/payloadMiddleware';

function injectLogger() {
	let isHandlingError = false;
	const ReactTryCatchBatchingStrategy = {
		// this is part of the BatchingStrategy API. simply pass along
		// what the default batching strategy would do.
		get isBatchingUpdates() {
			return ReactDefaultBatchingStrategy.isBatchingUpdates;
		},

		batchedUpdates(...args) {
			try {
				ReactDefaultBatchingStrategy.batchedUpdates(...args);
			} catch (e) {
				if (isHandlingError) {
					// our error handling code threw an error. just throw now
					throw e;
				}

				isHandlingError = true;
				try {
					TraceKit.report(e);
				} finally {
					isHandlingError = false;
				}
			}
		},
	};

	ReactUpdates.injection.injectBatchingStrategy(ReactTryCatchBatchingStrategy);
}

export default function initReactLogger({ serverUrl, getState, processState = state => state }) {
	initErrorTransformer(serverUrl, {
		payloadMiddleware: getStatePayloadMiddleware(() => processState(getState())),
	});
	injectLogger();
}
