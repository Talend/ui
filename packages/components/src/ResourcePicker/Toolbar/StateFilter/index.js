import StateFilter, { TYPES } from './StateFilter.component';

export default StateFilter;
export { TYPES };
