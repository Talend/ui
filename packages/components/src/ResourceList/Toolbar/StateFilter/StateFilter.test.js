import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import StateFilter, { TYPES } from './StateFilter.component';

describe('StateFilter', () => {
	it('should render', () => {
		const props = {
			favorites: false,
			certified: false,
			onChange: () => {},
			types: [TYPES.CERTIFIED, TYPES.FAVORITES],
		};

		const { container } = render(<StateFilter {...props} />);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render with states to set to true', () => {
		const props = {
			favorites: true,
			certified: true,
			onChange: () => {},
			types: [TYPES.CERTIFIED, TYPES.FAVORITES],
		};

		render(<StateFilter {...props} />);
		expect(screen.getByLabelText('Certified')).toHaveAttribute('aria-pressed', 'true');
		expect(screen.getByLabelText('Favorites')).toHaveAttribute('aria-pressed', 'true');
	});

	it('should render with only favorites', () => {
		const props = {
			favorites: false,
			onChange: () => {},
			types: [TYPES.FAVORITES],
		};

		render(<StateFilter {...props} />);
		expect(screen.getByLabelText('Favorites')).toHaveAttribute('aria-pressed', 'false');
		expect(screen.queryByLabelText('Certified')).not.toBeInTheDocument();
	});

	it('should render with only certified', () => {
		const props = {
			certified: false,
			onChange: () => {},
			types: [TYPES.CERTIFIED],
		};

		render(<StateFilter {...props} />);
		expect(screen.getByLabelText('Certified')).toHaveAttribute('aria-pressed', 'false');
		expect(screen.queryByLabelText('Favorites')).not.toBeInTheDocument();
	});
	it('should trigger onChange callback with the new state on click', async () => {
		const user = userEvent.setup();

		const onChange = jest.fn();
		render(
			<StateFilter onChange={onChange} types={[TYPES.FAVORITES, TYPES.CERTIFIED]} certified />,
		);
		expect(onChange).not.toHaveBeenCalled();

		await user.click(screen.getByLabelText('Certified'));
		expect(onChange).toHaveBeenCalledWith(TYPES.CERTIFIED, false);

		await user.click(screen.getByLabelText('Favorites'));
		expect(onChange).toHaveBeenCalledWith(TYPES.FAVORITES, true);
	});
});
