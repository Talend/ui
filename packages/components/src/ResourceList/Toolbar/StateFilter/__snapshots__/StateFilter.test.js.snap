// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateFilter should render 1`] = `
<div
  class="tc-resource-picker-state-filters theme-tc-resource-picker-state-filters"
>
  <span
    class="theme-option-label"
  >
    Filter:
  </span>
  <button
    aria-describedby="42"
    aria-label="Certified"
    aria-pressed="false"
    class="theme-tc-resource-picker-certified-filter tc-icon-toggle theme-tc-icon-toggle btn btn-link"
    type="button"
  >
    <span
      class="CoralIcon"
      name="talend-badge"
    />
  </button>
  <button
    aria-describedby="42"
    aria-label="Favorites"
    aria-pressed="false"
    class="theme-tc-resource-picker-favorite-filter tc-icon-toggle theme-tc-icon-toggle btn btn-link"
    type="button"
  >
    <span
      class="CoralIcon"
      name="talend-star"
    />
  </button>
</div>
`;
